libs/ballot-interpreter-vx/src/hmpb/votes.ts

```diff
@@ -1,58 +1,52 @@
 import {
-  AnyContest,
-  Candidate,
-  CandidateContest,
+  CandidateId,
   CandidateVote,
-  MsEitherNeitherContest,
+  ContestId,
+  Election,
   VotesDict,
-  YesNoContest,
+  WriteInCandidate,
   YesNoOption,
+  YesNoOptionId,
   YesNoVote,
 } from '@votingworks/types';
-import { inspect } from 'util';
+import { assert, find, throwIllegalValue } from '@votingworks/utils';
 
 export function addVote(
+  election: Election,
   votes: VotesDict,
-  contest: CandidateContest,
-  candidate: Candidate
+  contestId: ContestId,
+  option: CandidateId | 'yes' | 'no' | YesNoOptionId | WriteInCandidate
 ): void;
 export function addVote(
+  election: Election,
   votes: VotesDict,
-  contest: YesNoContest,
-  yesNo: 'yes' | 'no'
-): void;
-export function addVote(
-  votes: VotesDict,
-  contest: MsEitherNeitherContest,
-  option: YesNoOption
-): void;
-export function addVote(
-  votes: VotesDict,
-  contest: AnyContest,
-  candidateOrYesNoOrEitherNeither: Candidate | 'yes' | 'no' | YesNoOption
+  contestId: ContestId,
+  option:
+    | WriteInCandidate
+    | 'yes'
+    | 'no'
+    | YesNoOption
+    | CandidateId
+    | YesNoOptionId
```

Comment:
I'm a little confused why you can use either `'yes' | 'no'` or `YesNoOption` or `YesNoOptionId` - seems like some of those should be the same? I noticed a change lower down that set the type of `optionId` to `'yes' | 'no'`